core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -212,8 +212,7 @@ public static List<String> getStringList(String property, JsonNode node) {
   }
 
   public static Set<String> getStringSet(String property, JsonNode node) {
-    Preconditions.checkArgument(
-        node.hasNonNull(property), "Cannot parse missing set: %s", property);
+    Preconditions.checkArgument(node.has(property), "Cannot parse missing set: %s", property);
```

Comment:
Why relax the constraint? I think it would be safe to do this because of the `isNull` check in the array handler, but I see no reason to do this.

Reply:
the null check is already being handled inside the array handler. Relaxing this one here allows for being more precise on the error reporting as we don't want to say `Cannot parse missing set: items` when in fact `items` was given, but it was null. What we'd want to report is `"Cannot parse from non-array value: items: null"`. I also added some tests for this to make sure we're reporting the right thing